File: Chapter9/Java/Writing Methods that Return a Value in Java/Calculator.java
// Calculator.java - This program performs arithmetic, ( +. -, *. /, % ) on two numbers
// Input:  Interactive.
// Output:  Result of arithmetic operation

import java.util.Scanner;

public class Calculator {
    public static void main(String args[]) {
        double numberOne, numberTwo;
        String numberOneString, numberTwoString;
        String operation;
        double result = 0;
        Scanner input = new Scanner(System.in);

        System.out.println("Enter the first number: ");
        numberOneString = input.nextLine();
        numberOne = Double.parseDouble(numberOneString);

        System.out.println("Enter the second number: ");
        numberTwoString = input.nextLine();
        numberTwo = Double.parseDouble(numberTwoString);

        System.out.println("Enter an operator (+.-.*,/,%): ");
        operation = input.nextLine();

        // Call performOperation method here
        result = performOperation(numberOne, numberTwo, operation);

        System.out.format("%.2f", numberOne);
        System.out.print(" " + operation + " ");
        System.out.format("%.2f", numberTwo);
        System.out.print(" = ");
        System.out.format("%.2f", result);

        System.exit(0);

    } // End of main() method.

    // Write performOperation method here.
    public static double performOperation(double num1, double num2, String opp) {

        if (opp.equals("*")) {
            return (num1 * num2);
        }
        if (opp.equals("/")) {
            return (num1 / num2);
        }
        if (opp.equals("+")) {
            return (num1 + num2);
        }
        if (opp.equals("-")) {
            return (num1 - num2);
        }
        return 0;
    }

} // End of Calculator class.

File: Chapter9/Python/Writing Functions that Return a Value in Python/Calculator.py
# Calculator.py - This program performs arithmetic, ( +. -, *. / ) on two numbers.
# Input:  Interactive
# Output:  Result of arithmetic operation

# Write performOperation function here

def performOperation(a, b, op):

    """ This function returns value after performing operation op on a and b """

    if op == '+':

        return a + b

    elif op == '-':

        return a - b

    elif op == '*':

        return a * b

    elif op == '/':

        return a / b;

if __name__ == '__main__':

    numberOne = int(input("Enter the first number: "))

    numberTwo = int(input("Enter the second number: "))

    operation = input("Enter an operator (+ - * /): ")

    # Call performOperation method here and store the value in "result"

    result = performOperation(numberOne, numberTwo, operation)

    print(str(numberOne) + " " + operation + " " + str(numberTwo) + " = " + str(result))

